src/state/slices/portfolioSlice/utils.ts

```diff
@@ -101,9 +102,11 @@ export const accountIdToLabel = (accountId: AccountSpecifier): string => {
   }
 }
 
-// note - this is not really a selector, more of a util
 export const accountIdToFeeAssetId = (accountId: AccountSpecifier): AssetId =>
-  chainIdToFeeAssetId(fromAccountId(accountId).chainId)
+  // the only way we get an accountId, is from a chainAdapter that supports that chain
+  // hence, a chainId obtained from an accountId is guaranteed to have a chain adapter
+  // and we can safely non-null assert that it will exist
```

Comment:
Be that as it may, we can't guarantee that this function isn't given a garbage `AccountSpecifier`. I'd prefer to see a check for a `ChainAdapter` before calling `getFeeAssetId()`, and throwing if we don't have one.

Reply:
That would go back to the original return type signature/ function definition committed in https://github.com/shapeshift/web/pull/2159/commits/99127ce1cc091fe62da1f881eb79635b11c2ddbc#diff-8095e05140b9b96f9b9f73c7254f6304d3ee27b080998907840d056aed8fe6d8R114 where we used optional chaining.
The rationale for not using optional chaining/early return/throwing and being optimistic is that in theory, we could call this with garbage `AccountId`s, but in effect, given the flow this is called with, it won't happen. 

If you look at the aforementioned commit that was reverted, you can see all the optional chaining we need to do just to make the type-system happy, which is why we've decided with @0xdef1cafe to revert it and go the "optimistic" route, which really isn't optimistic.

Reply:
Sounds like the lesser of two evils - I'm cool with that. Thanks for the explanation!